File: TypeScript/codeSignal/rotateMatriz.ts


function rotateMatriz(a: number[][]): number[][] {
  a.reverse()
  for (let i = 0; i < a.length; i++) {
    for (let j = 0; j < i; j++) {
      [a[i][j], a[j][i]] = [a[j][i], a[i][j]]
    }
  }
  return a
}
const a = [
  [1, 2, 3],
  [4, 5, 6],
  [7, 8, 9]
]

console.log(rotateMatriz(a))

File: Go/codeSignal/rotateMatriz.go
package codesignal

func RotateMatriz(a [][]int) [][]int {
	reverseRow(a)
	for i := 0; i < len(a); i++ {
		for j := 0; j < i; j++ {
			a[i][j], a[j][i] = a[j][i], a[i][j]
		}
	}
	return a
}

func reverseRow(row [][]int) {
	i, j := 0, len(row)-1
	for i < j {
		row[i], row[j] = row[j], row[i]
		i++
		j--
	}
}
